### ISSUE TITLE
Access parent stack outputs in custom cdk

### ISSUE THREAD

**Issue by oste** [aws-amplify/amplify-cli#9644]
### Is this feature request related to a new or existing Amplify category?

auth

### Describe the feature you'd like to request

I am following the instructions for adding custom resources using the cdk here https://docs.amplify.aws/cli/custom/cdk/. What I would like to do is update the unauth role that amplify added during auth creation. I would like to add an inline policy like this:

```
     const role = am.Role.fromRoleArn(
          this,
          'imported-role',
          `my-role-arn`
        ),

    const statements = new iam.PolicyStatement();
    statements.addActions('appsync:GraphQL');
    statements.addResources('arn:aws:appsync:*:*:apis/*/*');

    role.attachInlinePolicy(
      new iam.Policy(this, 'api-access', {
        statements: [statements],
      })
    );

```

The problem I am having is I have to hardcode the `my-role-arn` when importing the auth role. I don't have access to the `UnauthRoleName` or `UnauthRoleArn` outputs in the root stack. There isn't anything useful in amplify-dependent-resources.d.ts either. I can't even concat a string b/c the generated role has what seems to be a random 5 number string in the name/arn.

### Describe the solution you'd like

I would like to be able to access the auth `UnauthRoleArn` or `UnauthRoleName` in my custom cdk resource. This might mean adding it as an output from the auth stack or maybe there is some way to access the root/parent stack outputs in my custom cdk code.

As a last ditch alternative if I can recreate the random 5 digit number I can concat the string together using `AmplifyHelpers` outputs like this

```
`arn:aws:iam::${cdk.Stack.of(this).account}:role/amplify-${helper.project}-${helper.env}-${randomNumber}-unauthRole`
```

### Describe alternatives you've considered

recreating auth in cdk completely but I would like to avoid this route.

### Is this something that you'd be interested in working on?

- [ ] 👋 I may be able to implement this feature request
- [ ] ⚠️ This feature might incur a breaking change

**Comment by alucio-echen** [comment]
I am running into this exact same issue. Attempts to use the new project override also ended in failure since CDK constructs are not accessible in override.ts. It would be really nice to get clearer documentation on how to pass outputs / parameters around. 

This is definitely not proper but I managed to get it to work by importing the #current-cloud-backend/amplify-meta.json file and getting the value from there. I had to enable the resolveJsonModule in the tsconfig for the custom resource to be able to import the JSON via a import statement.
